### ISSUE TITLE
getScript docs have a possibly misleading warning about success resolving before execution

### ISSUE THREAD

**Issue by albell** [jquery/api.jquery.com#420]
The [getScript docs](http://api.jquery.com/jquery.getscript/) say about the success callback that:

"The callback is fired once the script has been loaded but not necessarily executed."

But in my testing that doesn't seem to be true. For a host page with:

```
var startTime = new Date();

$.getScript("test.js")
 .done(function( script, textStatus ) {
    console.log( textStatus );
    console.log( "Done callback executing now.")
  })
  .fail(function( jqxhr, settings, exception ) {
    console.log("error." );
});
```

loading the following "test.js" script which ties up the UI for 5 seconds:

```
console.log("ajaxed script starting to execute.");
var newTime = new Date();
while (newTime - startTime < 5000) {
    newTime = new Date();
}
console.log("elapsed time", newTime - startTime);
console.log("ajaxed script finished executing.");
```

results in the same predictable console output in both FF & Chrome:

```
ajaxed script starting to execute.
elapsed time 5000 
ajaxed script finished executing.
success
Done callback executing now. 
```

In other words, the success callback does not ever fire until the loaded script is _both loaded and executed_. This seems to be because in the [source](https://github.com/jquery/jquery/blob/master/src/ajax/script.js), the globalEval function is calling the script immediately:

```
converters: {
    "text script": function( text ) {
        jQuery.globalEval( text );
        return text;
    }
}
```

So are the docs wrong? If they are correct, then in what specific cases will the success callback fire _before_ the script is executed?

(question [moved from StackOverflow](http://stackoverflow.com/questions/21196617/jquery-getscript-load-vs-execution))

**Comment by dandv** [comment]
I was also tripped by this and spent a good half hour trying to figure out how to determine when exactly the callback was fired ([jsbin](http://jsbin.com/puyuyi/4/edit?html,js,console). Maybe the documentation should be amended to something like,

> The callback is fired once the script has been loaded and executed, except in some XXX-dependent cases when the script has only loaded but execution hasn't finished. Make sure test whether your external script falls into one of these categories.

**Comment by marioacc** [comment]
@dandv 

> Make sure to test whether your external script falls into one of these categories.

Do you have some example/insight about how to do that?
